airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -52,12 +56,23 @@ class ShopifyBulkManager:
     # attempt counter
     concurrent_attempt: int = field(init=False, default=0)
 
-    # default logger
-    logger: AirbyteLogger = field(init=False, default=logging.getLogger("airbyte"))
-
     # currents: job_id, job_state
     job_id: Optional[str] = field(init=False, default=None)
     job_state: ShopifyBulkStatus = field(init=False, default=None)
+    # Each job ideally should be executed within the specified time (in sec),
+    # to maximize the performance for multi-connection syncs and control the bulk job size within +- 7 mins (420 sec),
+    # Ideally the source will balance on it's own rate, based on the time taken to return the data for the slice.
+    job_elapsed_time_threshold_sec: float = field(init=False, default=420.0)
+    # 2 sec is set as default value to cover the case with the empty-fast-completed jobs
+    job_last_elapsed_time_sec: float = field(init=False, default=2.0)
+    current_job_elapsed_time: float = field(init=False, default=0.0)
+    job_should_expand_slice_size: bool = field(init=False, default=False)
+    slice_size_expand_factor: int = field(init=False, default=2)
+    job_should_reduce_slice_size: bool = field(init=False, default=False)
+    slice_size_reduce_factor: int = field(init=False, default=2)
```

Comment:
Can we set this as a constant to indicate that it never changes in the current flow?